core/shared/src/main/scala/scalaz/zio/Promise.scala

```diff
@@ -48,13 +46,12 @@ class Promise[E, A] private (private val state: AtomicReference[State[E, A]]) ex
       case Pending(_) => false
     })
 
-  /**
-   * Retrieves the value of the promise, suspending the fiber running the action
+  /** Retrieves the value of the promise, suspending the fiber running the action
    * until the result is available.
    */
-  final def await: IO[E, A] =
-    IO.effectAsyncInterrupt[Any, E, A](k => {
-      var result = null.asInstanceOf[Either[Canceler, IO[E, A]]]
+  final def await: IO[Any, E, A] =
```

Comment:
This is a typo, `IO` will only support 2 type parameters `E` and `A`, not the `R` type parameter. Same as below.